doc/xgboost_on_sqlflow.md

```diff
@@ -0,0 +1,139 @@
+# xgboost on sqlflow
+
```

Comment:
Add a few sentences summarizing the content of this doc. [ref](https://www.freecodecamp.org/news/how-to-write-a-good-software-design-document-66fcf019569c/)

> A high-level summary that every engineer at the company should understand and use to decide if it’s useful for them to read the rest of the doc. It should be 3 paragraphs max.

Reply:
@tonyyang-svail  I have reorganized the doc, splitting it into _overview_, _context_ and _proposed solution_.